packages/miniflare/src/plugins/assets/index.ts

```diff
@@ -195,10 +200,19 @@ const walk = async (dir: string) => {
 							`Ensure all assets in your assets directory "${dir}" conform with the Workers maximum size requirement.`
 					);
 				}
+				const modifiedTime = (await fs.stat(filepath)).mtimeMs;
 
-				manifest.push(
-					await hashPath(encodeFilePath(relativeFilepath, path.sep))
-				);
+				manifest.push({
+					pathHash: await hashPath(encodeFilePath(relativeFilepath, path.sep)),
+					// In prod the structure of each manifest entry is [header, pathHash, contentHash].
+					// In dev we want to avoid reading and hashing all file contents every time we
+					// reload the dev server, so in dev the 'content hash' is hash(path + modifiedTime).
+					// This way the 'content hash' still updates on file change, so the eTag also updates,
+					// and we don't incorrectly get a 304.
+					contentHash: await hashPath(
+						encodeFilePath(relativeFilepath, path.sep) + modifiedTime.toString()
```

Comment:
oh clever! I was going to hash the whole thing à la prod, but this saves an expensive computation!